filewatcher/filewatcher.go

```diff
@@ -104,20 +108,47 @@ func (r *Result) watcherLoop(
 	pollingInterval time.Duration,
 ) {
 	forceReload := func(mtime time.Time) {
+		isDir, err := isDirectory(path)
+		if err != nil {
+			logger.Log(context.Background(), "filewatcher: isDirectory error: "+err.Error())
+			return
+		}
 		f, err := limitopen.OpenWithLimit(path, softLimit, hardLimit)
```

Comment:
shouldn't you check `isDir` before doing this, and only do this if it's not a directory?

Reply:
added that check